fs-index/src/index.rs

```diff
@@ -13,8 +13,13 @@ use walkdir::{DirEntry, WalkDir};
 use log;
 
 use crate::{ArklibError, Result, ARK_FOLDER, INDEX_PATH};
-use data_resource::ResourceIdTrait;
-use data_resource::{Resource, ResourceId};
+use data_resource;
+
+//todo: this is not the way to go:
+#[cfg(feature = "hash_blake3")]
+use data_resource::blake3::ResourceId as ResourceId;
+#[cfg(not(feature = "hash_blake3"))]
+use data_resource::crc32::ResourceId as ResourceId;
```

Comment:
Yes, but the idea was to remove the unifying type from `data-resource`. Now, I think we should not define implementations for the `ResourceId` trait in production crates at all.

Reply:
Let me ensure I've got this right:
- `data-resource` only contains the definition for `ResourceIdTrait`.
- `dev-hash` defines various `ResourceId` structs that implement `ResourceIdTrait`, without pushing them behind feature flags.
- `fs-index` imports the required `ResourceId` from `dev-hash`, with this import being controlled by a feature flag.

If this is accurate, I see a potential issue:

We would need to apply the same compilation flag mechanism to other crates that also require `ResourceId`, such as `fs-metadata`, `fs-properties`, and `ark-cli`. 
So if we need to switch between a cryptographic hash and a non-cryptographic implementation of `ResourceId`, we'd have to manually modify code across all these locations.


What do you think?

Reply:
> Let me ensure I've got this right:
> * data-resource only contains the definition for `ResourceIdTrait`.

Yes, just a trait. Maybe some auxiliary functions later, but they would be built on top of the trait, abstracted from a particular hash function.

> * `dev-hash` defines various `ResourceId` structs that implement `ResourceIdTrait`, without pushing them behind feature flags.

`dev-hash` (tentative name) would contain several implementations for `ResourceIdTrait`. No conditional compilation, no feature flags. Just all hash functions in a single crate. I expect adding new hash functions there when we benchmark them.

> * `fs-index` imports the required `ResourceId` from `dev-hash`, with this import being controlled by a feature flag.

No, `fs-index` would be built on top of `ResourceIdTrait`. Same as `fs-storage` and all other crates. We actually can also move all benchmarks into `dev-hash`. In fact, `dev-hash` would represent first app, making `ark-cli` a second app (it could depend on `dev-hash`).